app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessagePresenter.kt

```diff
@@ -32,12 +32,17 @@ class MessagePresenter {
         gattClient.stopClient()
     }
 
-    fun convertToPreview(data: String): ArrayList<String> {
+    fun convertToPreview(data: String): Pair<Boolean, ArrayList<String>> {
+        var valid = true
         val list: ArrayList<String> = ArrayList()
         for (letter in data) {
-            list.add(DataToByteArrayConverter.CHAR_CODES.getValue(letter))
+            if (DataToByteArrayConverter.CHAR_CODES.containsKey(letter)) {
+                list.add(DataToByteArrayConverter.CHAR_CODES.getValue(letter))
+            } else {
+                valid = false
```

Comment:
break here, once you know it's invalid, no need to continue

Reply:
Done.

Reply:
@iamareebjamal we cannot break here. If the string is like "^efjjih" and "^" is the unimplemented character, rest of the characters also won't be displayed if break is placed in the else section. We have to allow the loop to run through the whole string such that the implemented characters can at least be displayed.